src/sdl/sdl_system.c

```diff
@@ -144,7 +144,16 @@ static ALLEGRO_SYSTEM *sdl_initialize(int flags)
    ALLEGRO_SYSTEM_SDL *s = al_calloc(1, sizeof *s);
    s->system.vt = vt;
 
-   SDL_Init(SDL_INIT_EVERYTHING);
+   // TODO: map allegro flags to sdl flags.
+   unsigned int sdl_flags = SDL_INIT_EVERYTHING;
+#ifdef __EMSCRIPTEN__
+   // SDL currently does not support haptic feedback for emscripten.
+   sdl_flags -= SDL_INIT_HAPTIC;
```

Comment:
That looks very odd... I'd rather see this as `sdl_flags &= ~SDL_INIT_HAPTIC;`

Reply:
Is it possible to do this by trying to init with `SDL_INIT_EVERYTHING` and falling back to `SDL_INIT_EVERYTHING & ~SDL_INIT_HAPTIC` if it fails? The reason being that, if SDL ever does implement a haptic driver for Emscripten, it will work without having to change this code back.

Reply:
@pedro-w Good idea. There is a Web api for [vibration](https://developer.mozilla.org/en-US/docs/Web/API/Vibration_API), so it's certainly possible that one day the haptic system will work. I'll try your backing-off strategy this evening.

Reply:
There seemed to be some sound issues when I did the backoff approach. I'm not confident that SDL truly cleans up everything when it shutsdown its systems.